packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -509,7 +511,8 @@ class ServiceExtensionManager {
         await _maybeAddServiceExtension(extension);
       }
 
-      if (_pendingServiceExtensions.isEmpty) {
+      if (_pendingServiceExtensions.isEmpty &&
```

Comment:
was this causing problems or was this defensive programming to avoid possibly throwing an  exception?
Likely there is a logic error somewhere if this is actually required.